src/compile/unit.ts

```diff
@@ -156,10 +156,13 @@ export class UnitModel extends ModelWithField {
       if (isFieldDef(channelDef)) {
         fieldDef = channelDef;
         specifiedScale = channelDef.scale;
+      } else if (isConditionalDef(channelDef) && isFieldDef(channelDef.condition)) {
+        fieldDef = channelDef.condition;
+        specifiedScale = channelDef.condition.scale;
```

Comment:
Yeah I tried this earlier too. However, TypeScript won't know that `channelDef.condition` is not only a FieldDef, but also a `ScaleFieldDef`.  So we either have to do type casting or have to do this and I generally prefer to avoid casting.